src/main/java/org/apache/sysds/runtime/instructions/cp/ParamservBuiltinCPInstruction.java

```diff
@@ -125,15 +127,24 @@ private void runFederated(ExecutionContext ec) {
 		System.out.println("[+] Running in federated mode");
 
 		// get inputs
-		PSFrequency freq = getFrequency();
-		PSUpdateType updateType = getUpdateType();
-		PSRuntimeBalancing runtimeBalancing = getRuntimeBalancing();
-		FederatedPSScheme federatedPSScheme = getFederatedScheme();
 		String updFunc = getParam(PS_UPDATE_FUN);
 		String aggFunc = getParam(PS_AGGREGATION_FUN);
+		PSUpdateType updateType = getUpdateType();
+		PSFrequency freq = getFrequency();
+		FederatedPSScheme federatedPSScheme = getFederatedScheme();
+		PSRuntimeBalancing runtimeBalancing = getRuntimeBalancing();
+		boolean weighing = getWeighing();
+		int seed = getSeed();
+
+		LOG.info("[+] Update Type: " + updateType);
+		LOG.info("[+] Frequency: " + freq);
+		LOG.info("[+] Data Partitioning: " + federatedPSScheme);
+		LOG.info("[+] Runtime Balancing: " + runtimeBalancing);
+		LOG.info("[+] Weighing: " + weighing);
+		LOG.info("[+] Seed: " + seed);
 
 		// partition federated data
-		DataPartitionFederatedScheme.Result result = new FederatedDataPartitioner(federatedPSScheme)
+		DataPartitionFederatedScheme.Result result = new FederatedDataPartitioner(federatedPSScheme, seed)
 				.doPartitioning(ec.getMatrixObject(getParam(PS_FEATURES)), ec.getMatrixObject(getParam(PS_LABELS)));
 		List<MatrixObject> pFeatures = result._pFeatures;
 		List<MatrixObject> pLabels = result._pLabels;
```

Comment:
Do you need to put pFeatures and pLabels in variables? It looks like it is only used once and much later in the method. The method could be shorter if you take pFeatures and pLabels directly from the result object.